gather_gitops.sh

```diff
@@ -0,0 +1,177 @@
+#!/bin/bash
+BASE_COLLECTION_PATH="/must-gather"
+
+GITOPS_COLLECTION_PATH="$BASE_COLLECTION_PATH/cluster-gitops"
+gitops_folder="$GITOPS_COLLECTION_PATH/gitops"
+
+# terminal text output color
+green='\033[0;32m'!=
+color_reset='\033[0m' 
+
+function getNamespaces() {
+  local result
+  local namespaces
+  local default="openshift-gitops"
+  local clusterScopedInstances
+  clusterScopedInstances=$(oc get subs openshift-gitops-operator -n openshift-operators -o json | jq '.spec.config.env[]|select(.name=="ARGOCD_CLUSTER_CONFIG_NAMESPACES").value' | tr -d '",')
+  if [[ "$(oc get subs openshift-gitops-operator -n openshift-operators -o jsonpath='{.spec.config.env}')" == "" ]]; then
+    namespaces=${default}
+  elif [[ "${clusterScopedInstances}" != "" ]]; then
+    if [[ "$(oc get subs openshift-gitops-operator -n openshift-operators -o json | jq '.spec.config.env[]|select(.name=="DISABLE_DEFAULT_ARGOCD_INSTANCE").value')" != "false" ]]; then
+      namespaces+="${clusterScopedInstances} ${default}"
+    else
+      namespaces="${clusterScopedInstances}"
+    fi
+  else 
+    echo "no namespaces found"
+  fi
+
+  local argocdInstances
+  argocdInstances=$(oc get ArgoCD --all-namespaces -o jsonpath='{.items[*].metadata.namespace}')
+
+  local total
+  total="${namespaces} ${argocdInstances}"
+  
+  export NAMESPACES=$(echo "${total}" | tr ' ' '\n' | sort -u)
+}
+
+function main() {
+  
+  getNamespaces
+
+  echo -e "${green}Starting GitOps Operator must-gather script..."
+  mkdir -p "$gitops_folder"
```

Comment:
Don't assume the folder can be created in all cases. It might be already present (is this a problem?) or the script might don't have write access.

Reply:
The whole file system is created new every time the command runs, so it shouldn't run into a problem where a file already exists. As for the write access, I believe that maybe the base image of openshift must-gather provides the write access? I'm not entirely sure, but I haven't seen any other must-gathers that explicitly provide write access

Reply:
If that's the case then you are right ;)

I was under the impression this dir is created locally in your laptop. Otherwise how are the logs stored in your PC? Maybe the wrapper must-gather does that already. Anyway, feel free to ignore this.